maggy/maggy/project_routing.py

```
@@ -0,0 +1,138 @@
+        meta=dict(raw.get("_meta") or {}),
+    )
+
+
+def load_project_profile(cwd: str, base: Path | None = None) -> ProjectProfile | None:
+    """Load a project's profile, or None if it has none / is unreadable."""
+    path = project_profile_path(cwd, base)
+    if not path.exists():
+        return None
+    try:
+        raw = yaml.safe_load(path.read_text()) or {}
+    except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)